### DISCUSSION TITLE
Difference between micro_train_batch_size & micro_rollout_batch_size

### DISCUSSION THREAD

**Discussion by pratikkumar018** [OpenRLHF/OpenRLHF#210] [General]
Can you please explain the difference between micro_train_batch_size & micro_rollout_batch_size. I tried going through the code and found micro_train_batch_size  to be used here.  
[https://github.com/OpenLLMAI/OpenRLHF/blob/main/openrlhf/trainer/ppo_utils/replay_buffer.py#L152](url) 
I am not sure  if these two values should be same everytime?

**Comment by hijkzzz** [comment]
micro_train_batch_size is the training batch size per gpu
micro_rollout_batch_size is the inference batch size per gpu for PPO

**Reply by pratikkumar018** [reply]
So when you say inference, does it mean batch_size during validation time or does it mean inference batch size for initial_model & reward model (frozen models)?

**Reply by wuxibin89** [reply]
@pratikkumar018 micro_rollout_batch_size is the size for sequence generation and initial_model/reward_model inference afterwards.
![image](https://github.com/OpenLLMAI/OpenRLHF/assets/10410189/51e93608-8f6f-475d-8634-03b3e98c98b3)

**Reply by pratikkumar018** [reply]
Thanks a lot
